wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -183,6 +188,10 @@ public WorkspaceRuntimes(
     workspaceRuntimesId = NameGenerator.generate("runtimes", 16);
   }
 
+  public static boolean isNamespaceNameValid(String name) {
```

Comment:
Personally I don't like introducing such **static** method here... Is really namespace name is related to `WorkspaceRuntimes`? I assume with static method you avoided cyclic dependency but maybe there is another way to do it. Maybe even all methods related to infrastructure namespace may be extracted from here? This class already contains 1083 lines which is pretty more than enough )

Reply:
Yep, I sort of slapped it on the first place that I saw was reachable from all call sites. Will try to find a better place for it.